distributed/tests/test_utils_test.py

```diff
@@ -474,6 +474,24 @@ def test_assert_story_malformed_story(story_factory):
         assert_story(story, [])
 
 
+@pytest.mark.parametrize("strict", [True, False])
+@gen_cluster(client=True, nthreads=[("", 1)])
+async def test_assert_story_identity(c, s, a, strict):
+    f1 = c.submit(inc, 1, key="f1")
+    f2 = c.submit(inc, f1, key="f2")
+    assert await f2 == 3
+    scheduler_story = s.story(f2.key)
+    assert scheduler_story
+    worker_story = a.story(f2.key)
+    assert worker_story
+    assert_story(worker_story, worker_story, strict=strict)
+    assert_story(scheduler_story, scheduler_story, strict=strict)
+    with pytest.raises(AssertionError):
+        assert_story(scheduler_story, worker_story, strict=strict)
+    with pytest.raises(AssertionError):
+        assert_story(worker_story, scheduler_story, strict=strict)
```

Comment:
I love this test :)

Reply:
The intent is super clear.  It seems very robust to me.  It also seems fairly powerful (it's not an easy test to pass)